python/foxglove-sdk-examples/ws-param-server/main.py

```diff
@@ -0,0 +1,97 @@
+"""
+This implements a parameter server for live visualization.
+
+View and edit parameters from a Parameters panel in Foxglove:
+https://docs.foxglove.dev/docs/visualization/panels/parameters
+"""
+
+import logging
+from typing import Optional
+import foxglove
+import time
+
+
+from foxglove import Capability
+from foxglove import ParameterType, ParameterValue, Parameter
+
+
+class ParameterStore(foxglove.ServerListener):
+    def __init__(self, parameters: list[Parameter]) -> None:
+        self.parameters = parameters
+
+    # Foxglove server callback
+    def on_get_parameters(
+        self,
+        client: foxglove.Client,
+        param_names: list[str],
+        request_id: Optional[str] = None,
+    ) -> list[Parameter]:
+        logging.debug(f"on_get_parameters: {param_names}, {client.id}, {request_id}")
+        return self.parameters
+
+    def on_set_parameters(
+        self,
+        client: foxglove.Client,
+        parameters: list[foxglove.Parameter],
+        request_id: str | None = None,
+    ) -> list[foxglove.Parameter]:
+        logging.debug(f"on_set_parameters: {parameters}, {client.id}, {request_id}")
+        existing_names = [p.name for p in self.parameters]
```

Comment:
I think self.parameters should be a dict. See the rust example.

If you wanted to keep this as is you could use a set or dict comprehension.

Using list everywhere is simple but inefficient. I know it's only an example, but I like setting a good example.

Reply:
I originally wrote it with a dict, but thought that it got in the way of understanding the interface. I suppose having a lot of parameters could be typical; I'll update.